openpype/tools/publisher/window.py

```diff
@@ -455,8 +480,65 @@ def _on_show_timer(self):
             self._reset_on_show = False
             self.reset()
 
-    def save_changes(self):
-        self._controller.save_changes()
+    def _checks_before_save(self, explicit_save):
+        """Save of changes may trigger some issues.
+
+        Check if context did change and ask user if he is really sure the
+        save should happen. A dialog can be shown during this method.
+
+        Args:
+            explicit_save (bool): Method was called when user explicitly asked
+                for save. Value affects shown message.
+
+        Returns:
+            bool: Save can happen.
+        """
+
+        if not self._controller.host_context_has_changed:
+            return True
+
+        title = "Host context changed"
+        if explicit_save:
+            message = (
+                "Context has changed since Publisher window was refreshed last"
+                " time.\n\nAre you sure you want to save changes?"
+            )
+        else:
+            message = (
+                "Your action requires save of changes but context has changed"
+                " since Publisher window was refreshed last time.\n\nAre you"
+                " sure you want to continue and save changes?"
+            )
```

Comment:
I've never gotten these messages - when should I be seeing these?

Reply:
Was able to get the message - nice! :) 

In a demo video of Nuke + new publisher it was shown to after publishing tweak the instances to "save it" for the next time you'll run the publish. With this PR you can still after publishing go to the Publish tab, tweak the state of instances but there's no save button. There's no save button, but CTRL+S does still work (that should be disabled then too?) and worse when clicking CTRL+S it now also shows a context changed message because my scene was incrementally saved during publish - adding to the confusion. :)

I'm fine with it not being possible to save until a reset happened but I'd have expected I shouldn't be able to change the active state on instances then either nor that CTRL+S should be possible.

Reply:
> that should be disabled then too?

Fixed

> I shouldn't be able to change the active state on instances

That is possible?

Reply:
> That is possible?

Sure was. After Publishing go to Publish tab.

Reply:
Not anymore

Reply:
Seems to work for me! :)

Reply:
Actually here's a fun one. :) You can still toggle the instance state.
Go to this menu mode:
![afbeelding](https://user-images.githubusercontent.com/2439881/227041794-6196a839-cfd1-4293-9e7c-44fd06825a0b.png)

Now select the instance, press **Spacebar**.
Haha, not a big one however since I can't save the state at that point in time.